src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager3.cs

```diff
@@ -0,0 +1,434 @@
+using Standard;
+using Microsoft.Win32;
+using System.Collections;
+using System.Collections.Generic;
+using System.Windows.Interop;
+using System.Windows.Media;
+using System.Windows.Appearance;
+using System.Diagnostics;
+
+
+namespace System.Windows;
+
+internal static class ThemeManager3
+{
+
+
+    #region Internal Methods
+
+    internal static void OnSystemThemeChanged()
+    {
+        if(IsFluentThemeEnabled)
+        {
+            IgnoreAppResourcesChange = true;
+
+            bool useLightColors = GetUseLightColors(Application.Current.ThemeMode);
+            var fluentThemeResourceUri = GetFluentThemeResourceUri(useLightColors);
+            AddOrUpdateThemeResources(Application.Current.Resources, fluentThemeResourceUri);
+
+            foreach(Window window in Application.Current.Windows)
+            {
+                if(window.ThemeMode == ThemeMode.None)
+                {
+                    ApplyStyleOnWindow(window, useLightColors);
+                }
+                else
+                {
+                    ApplyFluentOnWindow(window);
+                }
+            }
+
+            IgnoreAppResourcesChange = false;
+        }
+        else
+        {
+            foreach(Window window in FluentEnabledWindows)
+            {
+                if(window == null || window.IsDisposed) continue;
+
+                if(window.ThemeMode == ThemeMode.None)
+                {
+                    RemoveFluentFromWindow(window);
```

Comment:
Every `RemoveFluentFromWindow` will remove the application's reource, is that right?

Reply:
No, that's wrong, while doing a replace I added the wrong one here. Pushing new changes.